Documentation/upgrading.md

```diff
@@ -1,173 +1,143 @@
+Check the current Kubernetes version.
 
-## Control Plane
+```sh
+$ kubectl version
+Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
+Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.6+coreos.1", GitCommit:"42a5c8b99c994a51d9ceaed5d0254f177e97d419", GitTreeState:"clean", BuildDate:"2017-06-21T01:10:07Z", GoVersion:"go1.7.6", Compiler:"gc", Platform:"linux/amd64"}
+```
+
+```sh
+$ kubectl get nodes
+NAME                               STATUS    AGE       VERSION
+node1.example.com                  Ready     21d       v1.6.6+coreos.1
+node2.example.com                  Ready     21d       v1.6.6+coreos.1
+node3.example.com                  Ready     21d       v1.6.6+coreos.1
+node4.example.com                  Ready     21d       v1.6.6+coreos.1
+```
 
-### kube-apiserver
+## Control Plane
 
-Edit the kube-apiserver daemonset. Change the container image name to `quay.io/coreos/hyperkube:v1.4.3_coreos.0`.
+Show the control plane daemonsets and deployments which will need to be updated.
 
-    $ kubectl edit daemonset kube-apiserver -n=kube-system
+```sh
+$ kubectl get daemonsets -n=kube-system
+NAME                             DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE-SELECTOR                     AGE
+kube-apiserver                   1         1         1         1            1           node-role.kubernetes.io/master=   21d
+kube-etcd-network-checkpointer   1         1         1         1            1           node-role.kubernetes.io/master=   21d
+kube-flannel                     4         4         4         4            4           <none>                            21d
+kube-proxy                       4         4         4         4            4           <none>                            21d
+pod-checkpointer                 1         1         1         1            1           node-role.kubernetes.io/master=   21d
+
+$ kubectl get deployments -n=kube-system
+kube-controller-manager           2         2         2            2           21d
+kube-dns                          1         1         1            1           21d
+kube-scheduler                    2         2         2            2           21d
+```
 
-Since daemonsets don't yet support rolling, manually delete each apiserver one by one and wait for each to be re-scheduled.
+### kube-apiserver
 
-    $ kubectl get pods -n=kube-system
-    # WARNING: Self-hosted Kubernetes is still new and this may fail
-    $ kubectl delete pod kube-apiserver-s62kb -n=kube-system
+Edit the `kube-apiserver` daemonset to rolling update the apiserver.
 
-If you only have one, your cluster will be temporarily unavailable. Remember the Hyperkube image is quite large and this can take a minute.
+```sh
+$ kubectl edit daemonset kube-apiserver -n=kube-system
+```
```

Comment:
Might be useful to document just bumping the image if no other changes are needed:

```
kubectl set image daemonset/kube-apiserver kube-apiserver=quay.io/coreos/hyperkube:v1.6.7_coreos.0
```

Reply:
Added